Source/WebKit/NetworkProcess/storage/FileSystemStorageManager.cpp

```diff
@@ -165,23 +165,56 @@ Expected<WebCore::FileSystemHandleIdentifier, FileSystemStorageError> FileSystem
     return createHandle(connection, FileSystemStorageHandle::Type::Directory, String { m_path }, { }, true);
 }
 
-bool FileSystemStorageManager::acquireLockForFile(const String& path, WebCore::FileSystemHandleIdentifier identifier)
+// https://fs.spec.whatwg.org/#file-entry-lock-take
+bool FileSystemStorageManager::acquireLockForFile(const String& path, WebCore::FileSystemHandleIdentifier identifier, LockType lockType)
 {
-    if (m_lockMap.contains(path))
+    auto iterator = m_lockMap.ensure(path, [identifier] -> Lock {
+        return { .identifier = identifier };
+    }).iterator;
+
+    ASSERT(iterator->value.identifier == identifier);
+    if (iterator->value.identifier != identifier)
```

Comment:
This seems not right. It is legit to create two `FileSystemStorageHandle`s with the same path (e.g. when [getFileHandle](https://fs.spec.whatwg.org/#api-filesystemdirectoryhandle-getfilehandle) is invoked twice, two handles will be created), and seems they should be able to create writable at the same time according to spec.